presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -387,6 +387,13 @@ public void testMapSubscript()
         assertQuery("SELECT map(array[(1,2)], array['a'])[(1,2)]", "SELECT 'a'");
     }
 
+    @Test
```

Comment:
might be nice to add a test for the non-numeric case that throws a NON_NUMERIC_SAMPLE_PERCENTAGE exception.

Reply:
There is already a test at TestAnalyzer.java - [link](https://github.com/prestodb/presto/blob/27d2234bf4972293667019d15121cf6cb3d14e95/presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestAnalyzer.java#L1308)